### ISSUE TITLE
gcc 4.6 override keyword

### ISSUE THREAD

**Issue by tjhei** [dealii/dealii#3839]
We still support gcc 4.6 but it doesn't support the override keyword (see http://stackoverflow.com/a/15776434), so I am getting compiler errors:
```
/ssd/deal-git/include/deal.II/lac/trilinos_sparse_matrix.h:2317:25: error: expected ‘;’ at end of member declaration
/ssd/deal-git/include/deal.II/lac/trilinos_sparse_matrix.h:2335:43: error: expected ‘;’ at end of member declaration
...
```

**Comment by tjhei** [comment]
``override`` is also only introduced in c++11. I imagine that we won't be able to build without c++11 now...

**Comment by bangerth** [comment]
We can just remove the keyword -- it is not necessary at all.

This came in via the recent linear operator patches. @tamiko and @jppelteret -- any reason not to just remove the keyword here again?

**Comment by bangerth** [comment]
Or alternatively to enclose these payload classes into `#ifdef DEAL_II_USE_CXX11`?

**Comment by masterleinad** [comment]
This is already enclosed by `#ifdef DEAL_II_USE_CXX11` (`lac/trilinos_sparse_matrix.h:2049:25`).
The only further C++11 feature I see so far is `TrilinosPayload & operator=(const TrilinosPayload &) = default;` in `include/deal.II/lac/trilinos_sparse_matrix.h:2207`. 
GCC 4.6 has only experimental support of C++11 but we recognize that this is sufficient for us to enable `DEAL_II_USE_CXX11`

I guess removing the overrides, the default copy assignment operator and `#ifdef DEAL_II_USE_CXX11` would be the easiest fix if we don't just say that GCC 4.6 doesn't support C++11.

**Comment by drwells** [comment]
Do we have a list of 'C++11 features that are not available in supported compilers' somewhere? I feel like we encounter similar problems to this frequently and every time I have to go dig these things up.
